src/node/desktop/src/main/toolbar-manager.ts

```diff
@@ -0,0 +1,160 @@
+/*
+ * toolbar-manager.ts
+ *
+ * Copyright (C) 2022 by RStudio, PBC
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+/* eslint-disable max-len */
+
+import { BrowserWindow } from 'electron';
+import * as fs from 'fs';
+
+export interface ToolbarData {
+  buttons: {
+    iconPath: string;
+    tooltip: string;
+    onClick: string;
+  }[];
+}
+
+export class ToolbarManager {
+  constructor() {}
+
+  async createToolbar(window: BrowserWindow, toolbarData: ToolbarData) {
+    const jsScript =
+      'console.log("windowOnload");' +
+      this.addToolbarJsAsText() +
+      ';' +
+      this.addStylesJsAsText() +
+      ';' +
+      this.addButtonsJsAsText(toolbarData);
+
+    try {
+      await window.webContents.executeJavaScript(jsScript);
+
+      // eslint-disable-next-line @typescript-eslint/no-implicit-any-catch
+    } catch (err: any) {
+      if (err.message !== 'An object could not be cloned.') {
+        console.log('Error message: ', err.message);
```

Comment:
Can we use the logging api here, instead of `console.log`? Look for examples of the form:

`logger().log*`